### ISSUE TITLE
Get grand time of all blocks in residual evaluation

### ISSUE THREAD

**Issue by YaqiWang** [idaholab/moose#11659]
## Rationale

@fdkong and me came up with this idea. When we do domain partitioning, we try to assign equal number of elements for all the partitions. This actually assumes the computing cost of all elements are the same. However, this is not true in reality. For example, fuel elements in a reactor typically require more computing time because there are fission events and additional evaluations are needed.

## Description

We can ask moose to evaluate the grand time of all blocks in a residual evaluation. The grand time is the cpu times on the blocks divided by the number of elements, which can be tallied in `subdomainChanged()`. The grand time can provide a good weighting used for domain partitioning. In this way, the load on all cores can be better balanced particularly for residual evaluation. This possibly can be done along with @friedmud 's performance log refactoring. We could create a vector postprocessor to do the time tally. If solver is better balanced in term of number of elements, the code can accept a factor to weight 1 and the grand time weighting.

## Impact
Better parallel performance for realistic applications.

**Comment by YaqiWang** [comment]
There are also costs associated with sides and boundaries. But the idea is to let MOOSE provide a DAG in term of computing cost, which can then be used by domain partitioners. We can gradually grow the capability along this line. I think grand times of blocks should be a good starting point. Considering that we have kernels, uos, auxkernels, materials, this will be an interesting task.

**Comment by friedmud** [comment]
Yep - we've been discussing this (timing over subdomains for repartitioning)... @fdkong will be looking into it.

**Comment by friedmud** [comment]
BTW: I don't know about the name "grand" time :-)

**Comment by YaqiWang** [comment]
oops. This sound grand hilarious.
